src/generators/minecraft-axolotl-character/MinecraftAxolotlCharacterGenerator.res

```diff
@@ -0,0 +1,1014 @@
+let requireImage = id => Generator.requireImage("./images/" ++ id ++ ".png")
+let requireTexture = id => Generator.requireImage("./textures/" ++ id ++ ".png")
+
+let id = "minecraft-axolotl-character"
+
+let name = "Minecraft Axolotl Character"
+
+let thumbnail: Generator.thumnbnailDef = {
+  url: Generator.requireImage("./thumbnail/v2thumbnail_256.jpeg"),
+}
+
+let imageIds = ["Background", "Folds", "Labels"]
+let toImageDef = (id): Generator.imageDef => {id: id, url: requireImage(id)}
+let images: array<Generator.imageDef> = imageIds->Js.Array2.map(toImageDef)
+
+let textures: array<Generator.textureDef> = [
+  {
+    id: "Skin",
+    url: requireTexture("Steve"),
+    standardWidth: 64,
+    standardHeight: 64,
+  },
+  {
+    id: "Blue",
+    url: requireTexture("axolotl_blue"),
+    standardWidth: 64,
+    standardHeight: 64,
+  },
+  {
+    id: "Cyan",
+    url: requireTexture("axolotl_cyan"),
+    standardWidth: 64,
+    standardHeight: 64,
+  },
+  {
+    id: "Pink",
+    url: requireTexture("axolotl_lucy"),
+    standardWidth: 64,
+    standardHeight: 64,
+  },
+  {
+    id: "Gold",
+    url: requireTexture("axolotl_gold"),
+    standardWidth: 64,
+    standardHeight: 64,
+  },
+  {
+    id: "Brown",
+    url: requireTexture("axolotl_wild"),
+    standardWidth: 64,
+    standardHeight: 64,
+  },
+]
+
+let steve = TextureMap.MinecraftCharacterLegacy.steve
+let alex = TextureMap.MinecraftCharacterLegacy.alex
+
+let script = () => {
+  // Define input textures
+  Generator.defineSelectInput("Skin Model Type", ["Steve", "Alex"])
+  Generator.defineTextureInput("Skin", {standardWidth: 64, standardHeight: 64, choices: []})
+  Generator.defineTextureInput(
+    "Tail Fin Texture",
+    {
+      standardWidth: 64,
+      standardHeight: 64,
+      choices: ["Blue", "Cyan", "Pink", "Gold", "Brown"],
+    },
+  )
+  Generator.defineTextureInput(
+    "Head Fins Texture",
+    {
+      standardWidth: 64,
+      standardHeight: 64,
+      choices: ["Blue", "Cyan", "Pink", "Gold", "Brown"],
+    },
+  )
+  // Define user variables
+  Generator.defineBooleanInput("Show Folds", true)
+  Generator.defineBooleanInput("Show Labels", true)
+  Generator.defineBooleanInput("Axolotl Eyes", true)
+  Generator.defineBooleanInput("Show Skin Overlay", true)
+
+  // Get user variable values
+  let alexModel = Generator.getSelectInputValue("Skin Model Type") === "Alex"
+  let showFolds = Generator.getBooleanInputValue("Show Folds")
+  let showLabels = Generator.getBooleanInputValue("Show Labels")
+  let showEyes = Generator.getBooleanInputValue("Axolotl Eyes")
+  let showOverlay = Generator.getBooleanInputValue("Show Skin Overlay")
+
+  // Head
+  Generator.drawTexture(
+    //Axolotl's right
+    "Skin",
+    (0, 8, 8, 8),
```

Comment:
Good work using the non-legacy functions 👍 

The `TextureMap` file provides all of the coordinates we need for the characters.

Let's start using those coordinates in all new Character generators.

If you look in the `TextureMap` file you'll see [this coordinate is already defined](https://github.com/pixelpapercraft/pixel-papercraft-generator-builder/blob/9e2a963985e764e7f589c1b31ac8f5be5f878969/src/generators/TextureMap.res#L203).

So for the head, you can now write this:

```
// Head
Generator.drawTexture("Skin", steve.base.head.right, (189, 214, 40, 40), ())
Generator.drawTexture("Skin", steve.base.head.front, (229, 214, 64, 40), ())
Generator.drawTexture("Skin", steve.base.head.left, (293, 214, 40, 40), ())
Generator.drawTexture("Skin", steve.base.head.back, (333, 214, 64, 40), ())
Generator.drawTexture("Skin", steve.base.head.top, (229, 174, 64, 40), ())
Generator.drawTexture("Skin", steve.base.head.bottom, (229, 254, 64, 40), ~flip=#Vertical, ())
```

Let me know how you go updating all of the coordinates to use the `TextureMap` file.

Reply:
Here's another tip that will make writing generators easier and less likely to have bugs.

For each face, you specify an "origin x" and an origin y" coordinate (named `ox` and `oy`)

Then you calculate all of the coordinates based on these two values.

For example, with the head, you can write this:

```
// Head
let ox = 229
let oy = 214
Generator.drawTexture("Skin", steve.base.head.right, (ox - 40, oy, 40, 40), ())
Generator.drawTexture("Skin", steve.base.head.front, (ox, oy, 64, 40), ())
Generator.drawTexture("Skin", steve.base.head.left, (ox + 64, oy, 40, 40), ())
Generator.drawTexture("Skin", steve.base.head.back, (ox + 64 + 40, oy, 64, 40), ())
Generator.drawTexture("Skin", steve.base.head.top, (ox, oy - 40, 64, 40), ())
Generator.drawTexture("Skin", steve.base.head.bottom, (ox, oy + 40, 64, 40), ~flip=#Vertical, ())
```

What do you think?

Reply:
well for both of those, I'm just kind of confused as to the benefits of using those over manually doing them as I did. having to calculate things based on an origin seems like it would be more liable to errors than just counting as I did, and I'd have to basically rewrite all the coordinates, which there are a lot of.

Reply:
[1] Using the `TextureMap` guarantees that all of the source coordinates are correct. There have been accidental mistakes in the past in generators. Also, this helps me when reviewing generators - I can see at a glance that the coordinates are accurate, nothing to check.

[2] There is a large benefit to using the `ox` and `oy` coordinate system for maintainability of the generators.

If we need to move the position of the head on the template, then you just need to update 2 numbers.

When we don't use the `ox` and `oy` idea, then we would need to update 12 numbers, and manually recalculate all of them. And that's just for one part.

I'd like for all future generators to use this strategy because the generator authors are often not the people who maintain them over time.